### PR TITLE
Communication service

### PR DESCRIPTION
**Pull request by jonasngs**

### CODE REVIEW

File: frontend/src/utils/configs.js [hunk 1]
```diff
@@ -2,5 +2,7 @@ export const USER_SVC_BASE_URL =
   window.__RUNTIME_CONFIG__.REACT_APP_URI_USER_SVC || "http://localhost:8000";
 export const MATCH_SVC_BASE_URL =
   window.__RUNTIME_CONFIG__.REACT_APP_URI_MATCH_SVC || "http://localhost:8001";
+export const COMMUNICATION_SVC_BASE_URL =
+  window.__RUNTIME_CONFIG__.REACT_APP_URI_MATCH_SVC || "http://localhost:8005";
```
Comment (Nijnxw) [inline frontend/src/utils/configs.js:6]: I think should be `window.__RUNTIME_CONFIG__.REACT_APP_URI_COMMUNICATION_SVC` right?

File: communication-service/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const cors = require("cors");
+const http = require("http");
+
+const app = express();
+
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+app.use(cors()); // config cors so that front-end can use
+app.options("*", cors());
+
```
Comment (Nijnxw) [inline communication-service/index.js:11]: Can you add a get endpoint with path `/`.

This is for the ingress to check if the server is live when we deploy on GCP.

File: communication-service/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const httpServer = http.createServer(app);
+const io = require("socket.io")(httpServer, {
+  cors: {
```
Comment (Nijnxw) [inline communication-service/index.js:14]: Can you add another attribute called `path: "/communication-api"` above line 14.

Used in deployment for frontend to differentiate the various services (as all the services has the same base url).

_Can reference matching-service in siqi's PR https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g8/pull/64_

File: frontend/src/components/ChatWindow/index.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+// import PropTypes from "prop-types";
+import "./styles.scss";
+import io from "socket.io-client";
+// import { COMMUNICATION_SVC_BASE_URL } from "../../utils/configs";
+
+import { useSelector } from "react-redux";
+import { userSelector } from "../../stores/user";
+import { matchSelector } from "../../stores/match/match.slice";
+
+const socket = io.connect("http://localhost:8005");
```
Comment (Nijnxw) [inline frontend/src/components/ChatWindow/index.js:11]: Following the changes in the communication-service/index.js file, need to add the `path` attribute like this:
```javascript
const socket = io.connect("http://localhost:8005", {
  path: "/communication-api",
});
```
Also, why did you decide not to use the `COMMUNICATION_SVC_BASE_URL` constant?

**Review by Nijnxw — CHANGES_REQUESTED** [review]

File: frontend/src/utils/configs.js (hunk 1 above)
Comment (jonasngs) [reply frontend/src/utils/configs.js]: fixed.

File: communication-service/index.js (hunk 2 above)
Comment (jonasngs) [reply communication-service/index.js:11]: fixed.

File: communication-service/index.js (hunk 3 above)
Comment (jonasngs) [reply communication-service/index.js]: fixed.

File: frontend/src/components/ChatWindow/index.js (hunk 4 above)
Comment (jonasngs) [reply frontend/src/components/ChatWindow/index.js]: fixed.

**Review by Nijnxw — APPROVED** [review]

**Pull request merged**